firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/serialization/CrashlyticsReportJsonTransform.java

```diff
@@ -0,0 +1,565 @@
+        case "platform":
+          builder.setPlatform(jsonReader.nextInt());
+          break;
+        case "installationUuid":
+          builder.setInstallationUuid(jsonReader.nextString());
+          break;
+        case "buildVersion":
+          builder.setBuildVersion(jsonReader.nextString());
+          break;
+        case "displayVersion":
+          builder.setDisplayVersion(jsonReader.nextString());
+          break;
+        case "session":
+          builder.setSession(parseSession(jsonReader));
+          break;
+        default:
+          jsonReader.skipValue();
+          break;
+      }
+    }
+    jsonReader.endObject();
+    return builder.build();
+  }
+
+  private static CrashlyticsReport.Session parseSession(JsonReader jsonReader) throws IOException {
+    final CrashlyticsReport.Session.Builder builder = CrashlyticsReport.Session.builder();
+
+    jsonReader.beginObject();
+    while(jsonReader.hasNext()) {
+      String name = jsonReader.nextName();
+      switch (name) {
+        case "generator":
+          builder.setGenerator(jsonReader.nextString());
+          break;
+        case "identifier":
+          builder.setIdentifierFromUtf8Bytes(Base64.decode(jsonReader.nextString(), Base64.NO_WRAP));
+          break;
+        case "startedAt":
+          builder.setStartedAt(jsonReader.nextLong());
+          break;
+        case "app":
+          builder.setApp(parseApp(jsonReader));
+          break;
+        case "os":
+          builder.setOs(parseOs(jsonReader));
+          break;
+        case "device":
+          builder.setDevice(parseDevice(jsonReader));
+          break;
+        case "events":
+          builder.setEvents(parseArray(jsonReader, CrashlyticsReportJsonTransform::parseEvent));
+          break;
+        default:
+          jsonReader.skipValue();
+          break;
+      }
+    }
+    jsonReader.endObject();
+
+    return builder.build();
+  }
+
+  private static CrashlyticsReport.Session.Application parseApp(JsonReader jsonReader) throws IOException {
+    final CrashlyticsReport.Session.Application.Builder builder = CrashlyticsReport.Session.Application.builder();
+
+    jsonReader.beginObject();
+    while(jsonReader.hasNext()) {
+      String name = jsonReader.nextName();
+      switch (name) {
+        case "identifier":
+          builder.setIdentifier(jsonReader.nextString());
+          break;
+        case "version":
+          builder.setVersion(jsonReader.nextString());
+          break;
+        case "displayVersion":
+          builder.setDisplayVersion(jsonReader.nextString());
+          break;
+        case "installationUuid":
+          builder.setInstallationUuid(jsonReader.nextString());
+          break;
+        default:
+          jsonReader.skipValue();
+          break;
+      }
+    }
+    jsonReader.endObject();
+
+    return builder.build();
+  }
+
+  private static CrashlyticsReport.Session.OperatingSystem parseOs(JsonReader jsonReader) throws IOException {
```

Comment:
"Os" you're killing me. I'd argue that in a line with `IOException`, we should be able to call this `parseOS`. :-)

But I guess you want to match it match the name in the object, so whatever.

Reply:
Yeah. :man_shrugging: